sync with master

repagh opened the pull request.

roryyorke commented:
D'oh, sorry, will fix this; has an extra WIP commit, and should be for your scikit-buildsystem branch (and ultimately, that PR).

Commit:
fix invalid variable usages from wrapper outputs

Commit:
delete unused mathematical.pyf

Commit:
fix math

Commit:
only return complex list when necessary, add unittest for mb05md

Commit:
remove var porm

Commit:
fix docstring of mb05md()

Commit:
clean up pep8 issues in mb05md

Commit:
clean more pep8 issues, fix mb05nd, add unittest for mb05nd

Commit:
fix CMakeLists, remove stub test_2

Commit:
make test_mb.py executable

Commit:
add unittests for mc01td, fix dp constraint

Commit:
fix typo in test_mb

Commit:
sg03ad: arange instead of range

Commit:
fix typos, remove obosolete suite function

Commit:
split test td04ad R and C method

Commit:
correct API call line for td04ad

Commit:
move td04ad static test

Commit:
replace np.tile with np.repeat wihtout calling np.shape

this fixes a fail of the test on powerpc architecture

Commit:
whitespace fixes

Commit:
remove unused module, clean pep8 warning

Commit:
fix python-control#347: missing exception attribute

bnavigator committed:
fix sg03ad test for continuos time and add discrete time test

bnavigator committed:
docstrings for sg03ad tests

repagh committed:
simplified conda build files

repagh committed:
build system simpler

bnavigator committed:
sg03ad N is not hidden

bnavigator committed:
add schur decomposition wrappers

bnavigator committed:
fix math.pyf for python2 error

bnavigator committed:
some cosmetics on the docstrings

repagh committed:
do not use pip for building the recipe; pip may install missing dependencies,
defeating the purpopse of a clean conda recipe

repagh committed:
simplify recipe, remove pip

repagh committed:
bring two conda recipes in line, one with openblas, one with mkl

bnavigator committed:
reorganize travis configuration, remove runtests.py

Linux builds:

conda
builds based on pip and Ubuntu 18.04 packages
python versions 2.7 to 3.8

OSX builds:

CONDA with custom SDK (not functional)
osx / pip packages

replaced old runtests.py with direct pytest call

bnavigator committed:
add libopenblas to run section

bnavigator committed:
bla_vendor openblas on CONDA=0

repagh committed:
Merge pull request #91 from bnavigator/work-on-build-system

Merging this, to have a base for fixing osx builds next

bnavigator committed:
fix SDK dir selection

bnavigator committed:
reorganize build matrix

bnavigator committed:
install all test files

repagh committed:
Merge pull request #93 from bnavigator/reorganize-travis

reorganize travis configuration, remove runtests.py
Wonderful. Great to see all variants build.

bnavigator committed:
Merge branch 'master' into fix-unused-vars

lytex committed:
add ab08nz function to support regular pencil for complex state-space systems

lytex committed:
add ab08n* tests (ab08nd executes successfully, ab08nz SEGFAULTs)

bnavigator committed:
clean __init__

bnavigator committed:
rework signatures for ab08nd and ab08nz

bnavigator committed:
rework analysis.ab08nz() function

bnavigator committed:
extend tests for ab08nX

bnavigator committed:
fix docstring

bnavigator committed:
remove complex warning in ab08nz

bnavigator committed:
switch to pytest for python-control on travis, combine coverage for coveralls

roryyorke committed:
Merge pull request #79 from bnavigator/fix-unused-vars

  - fixes and tests for mb05md, mb05nd, mc01td
  - removed unnecessary mathematical.pyf

bnavigator committed:
move the schur tests into test_mb.py

bnavigator committed:
add the test file to CMakeLists.txt

bnavigator committed:
remove test_mb03schur.py entry from CMakeLists.txt

bnavigator committed:
add dimensions to returned values in docstrings

bnavigator committed:
docstring fix of index and slice

bnavigator committed:
another slice in docstring fix

bnavigator committed:
readd tolerance calculation docstring, reformat lzwork docstring

roryyorke committed:
a further slice in docstring fix

bnavigator committed:
yet another docstring index fix

bnavigator committed:
map logical to 1 or 0 independent of compiler in mc01td

bnavigator committed:
move mc01td to its alphabetical place

bnavigator committed:
mc01td() docstring to numpydoc [skip ci]

(except first line for signature and no short summary)

bnavigator committed:
run examples in testsuite

bnavigator committed:
clean test files

bnavigator committed:
fix syntax errrors in test_sb.py

bnavigator committed:
test_sb10jd did not use the return values

bnavigator committed:
ease precision tolerance for sb10jd test
